ClosedXML/Excel/Ranges/XLRangeBase.cs

```diff
@@ -635,15 +635,19 @@ public XLCell FirstCellUsed(Boolean includeFormats, Func<IXLCell, Boolean> predi
                         if (!cell.IsEmpty(true)) return cell;
                     }
                 }
+            }
 
-                var intersectedRanges = Worksheet.MergedRanges.GetIntersectedRanges(RangeAddress).ToList();
-                if (intersectedRanges.Any())
-                {
-                    Int32 minRo = intersectedRanges.Min(r => r.RangeAddress.FirstAddress.RowNumber);
-                    Int32 minCo = intersectedRanges.Min(r => r.RangeAddress.FirstAddress.ColumnNumber);
+            var intersectedRanges = Worksheet.MergedRanges.GetIntersectedRanges(RangeAddress)
+                .Where(r => predicate == null || predicate(r.FirstCell())).ToList();
```

Comment:
I think we should add overloads to `GetIntersectedRanges` to accept a predicate which is passed through. Also the nullable predicate can be invoked with a shorter syntax:

```c#
            var intersectedRanges = Worksheet
                .MergedRanges
                .GetIntersectedRanges(RangeAddress, predicate?.Invoke(r.FirstCell()) ?? true);
```
And doesn't look like the `ToList()` is necessary.

Reply:
Oh the other hand, I see `Quandrant.GetIntersectedRanges` uses `yield return`, so passing through the predicate probably doesn't add much value? Or does it?

Reply:
Without `ToList()`, the collection will be enumerated twice (`.Any()` will break on the first item, but then there are two `Min()` calls) which is undesirable.

> `Quandrant.GetIntersectedRanges` uses `yield return`, so passing through the predicate probably doesn't add much value

I agree with this - the effect will be the same.

Reply:
>  Also the nullable predicate can be invoked with a shorter syntax:...

Whole 5 characters shorter :)  OK, will use it.

Reply:
It's not about the number of characters. It's (for me at least) just easier to understand.